common/djangoapps/student/tests/tests.py

```diff
@@ -246,16 +256,52 @@ def test_linked_in_url_not_exists_without_config(self):
             }
         )
 
-        # adding config. linked-in-url will be return
+        # Enabling the configuration will cause the LinkedIn
+        # "add to profile" button to appear.
         LinkedInAddToProfileConfiguration(
-            dashboard_tracking_code='0_mC_o2MizqdtZEmkVXjH4eYwMj4DnkCWrZP_D9',
-            enabled=True).save()
+            company_identifier='0_mC_o2MizqdtZEmkVXjH4eYwMj4DnkCWrZP_D9',
+            enabled=True
+        ).save()
 
         status_dict = _cert_info(user, course, cert_status, 'honor')
-        self.assertIn(
-            'http://www.linkedin.com/profile/add?_ed=0_mC_o2MizqdtZEmkVXjH4eYwMj4DnkCWrZP_D9',
-            status_dict['linked_in_url']
+        expected_url = (
+            'http://www.linkedin.com/profile/add'
+            '?_ed=0_mC_o2MizqdtZEmkVXjH4eYwMj4DnkCWrZP_D9&'
+            'pfCertificationName=edX+Verified+Certificate+for+Demo+Course&'
+            'pfCertificationUrl=http%3A%2F%2Fs3.edx%2Fcert&'
+            'source=o'
+        )
+        self.assertEqual(expected_url, status_dict['linked_in_url'])
+
+    @ddt.data(
+        ('honor', 'edX Honor Code Certificate for DemoX'),
+        ('verified', 'edX Verified Certificate for DemoX'),
+        ('professional', 'edX Professional Certificate for DemoX'),
+        ('default_mode', 'edX Certificate for DemoX')
+    )
+    @ddt.unpack
+    def test_linked_in_url_certificate_types(self, cert_mode, cert_name):
+        user = Mock(username="fred")
+        course = Mock(
+            display_name='DemoX',
+            end_of_course_survey_url='http://example.com',
+            certificates_display_behavior='end'
         )
+        cert_status = {
+            'status': 'downloadable',
+            'grade': '67',
+            'download_url': 'http://edx.org',
+            'mode': cert_mode
+        }
+
+        cache.clear()
```

Comment:
Why do we need to clear the cache here? Would it make sense to put this into a setup? Or to leave a comment explaining it for other people?